### DISCUSSION TITLE
Several questions about html, backend, frontend and widgets.

### DISCUSSION THREAD

**Discussion by twicemice** [TriliumNext/Trilium#3560] [Q&A]
Hi 
Thanks again @zadam  for your work. 
I need to resolve several doubts. 
First of all, I am working with widgets and I need to know if the size of the buttons can be increased. On the other hand, I don't know if there would be any other possibility to integrate the buttons inside the note, not in the surroundings. I have already tried all the possibilities described in the documentation, but the buttons remain outside the working area and in a small size. 
I ask this question because I have managed to make several widgets and I also have several render notes, but I can't find a way to combine the advantages and flexibility of html with the backend and frontend functionalities of the api that we find in widgets. Is there any way to combine both (html and api/widgets)?
Thanks again Zadam for your work.  
Thank you for your attention.

**Comment by zadam** [comment]
TBH I don't really understand your question. What do you mean by "buttons"? Screenshots might be helpful.

> Is there any way to combine both (html and api/widgets)?

This is also not clear to me. You mentioned render notes and now widgets - both are done with HTML, so of course you can combine both.

**Comment by twicemice** [comment]
Sorry, I'll try to explain it better
I have created different widgets that I add to certain notes. The functionality is simple: they include a button that creates a new note as well as opening a url in the browser. 
[https://github.com/zadam/trilium/discussions/3298]
![imagen](https://user-images.githubusercontent.com/116260332/215046448-c5d745b6-ba9b-4fbf-89b8-ddfd11c6c08d.png)

My problem is that I would like to include these buttons in the note itself and be able to give them a different appearance, especially the size. 

On the other hand I have created render notes that allow me to enjoy the flexibility of Html and small javascriprt functionalities, but I have not been able to access the API functions, as I can do with widgets. 

![imagen](https://user-images.githubusercontent.com/116260332/215046630-44c710e2-7191-4801-ac18-498c5b48b211.png)

This kind of note allows me to open a new browser tab, but not to create a new note. 

So my current interest is to combine the advantages of both methods, having html in the note whose javascript allows me to create/modify new notes through the API.  

Maybe  the frontend api  runOnBackend(script, params) could help?

Thanks

**Reply by zadam** [reply]
> My problem is that I would like to include these buttons in the note itself and be able to give them a different appearance, especially the size.

You can give them a different appearance. Widgets are using HTML, you can do anything you want there. Using these, you can also position the widget so it appears as if it was within the note:

```
get position() { return 100; } // higher value means position towards the bottom/right

get parentWidget() { return 'note-detail-pane'; }
```

> On the other hand I have created render notes that allow me to enjoy the flexibility of Html and small javascriprt functionalities, but I have not been able to access the API functions, as I can do with widgets.

JavaScript embedded within the HTML of rendered note doesn't have access to the API, that's correct.

However, you can create child JS frontend notes of this main HTML note and these will have access to the `api` object.

See "Weight tracker" in the [demo document](https://github.com/zadam/trilium/wiki/Document#demo-document) which uses this approach.

**Reply by twicemice** [reply]
Thank You very much!!
